tarodb/asset_minting.go

```diff
@@ -277,6 +279,19 @@ func (a *AssetMintingStore) CommitMintingBatch(ctx context.Context,
 
 		return nil
 	})
+
+	sqlError := ParseSqlError(err)
+	if sqlError != nil {
+		var errUnique *ErrSqlUniqueConstraintViolation
+		if errors.As(sqlError, &errUnique) && strings.Contains(
```

Comment:
I think we should add a helper function to the `sqlerrors.go` file that you can call as `IsErr(sqlError, &ErrSqlUniqueConstraintViolation{})`.

Reply:
`errUnique` is used in the `if` block. I don't think I can get an equivalent `errUnique` from `IsErr` without just recreating `errors.As`. Am I missing something?

Reply:
Ah, you're right... But maybe here `errors.Is(sqlError, &ErrSqlUniqueConstraintViolation{})` would be enough if we end up string matching on the message anyway? Combined with my other comment about including the original error I mean.

Reply:
As far as I understand, `sqlError` will never be equal to  `&ErrSqlUniqueConstraintViolation{}` unless we constrain ourselves to no fields in the struct `ErrSqlUniqueConstraintViolation`. Therefore, for `errors.Is` to work properly, we need to implement `Is` as a `ErrSqlUniqueConstraintViolation` method. How should I do that?

I can't compare on the `DbError` field so do I need a marker field or something?

```
type ErrSqlUniqueConstraintViolation struct {
	DbError error
}

func (e ErrSqlUniqueConstraintViolation) Is(target error) bool {
	?
}
```

Reply:
I thought `errors.Is()` looks at the type. But you're right, it doesn't and we'd need to implement `Is()` instead. Okay, I guess let's stick with the original pattern you had, just declaring the variable and using `errors.As()`.

Reply:
The alternative would be a simple type switch to implement `Is()`:
```
func (e ErrSqlUniqueConstraintViolation) Is(target error) bool {
	switch target.(type) {
	case *ErrSqlUniqueConstraintViolation:
		return true

	default:
		return false
	}
}
```

But that feels like overkill to me...

Reply:
I agree that is a bit inconvenient. Thanks for taking a look!